### PR TITLE
Add dtype when copy parameters

### PR DESCRIPTION
**Pull request by kice**
## Description ##
Fixes issue https://github.com/apache/incubator-mxnet/issues/13335

### CODE REVIEW

**Comment by stu1130** [comment]
@mxnet-label-bot add [pr-awaiting-review]
Thanks for your contribution @kice 
Could you please change the description to "Fixes #13335"?

**Comment by vandanavk** [comment]
@apeforest @anirudh2290 for review

File: src/c_api/c_predict_api.cc [hunk 1]
```diff
@@ -210,18 +210,22 @@ int _CreatePartialOut(const char* symbol_json_str,
...
-    arg_arrays.push_back(nd);
   }
   for (size_t i = 0; i < aux_shapes.size(); ++i) {
-    NDArray nd = NDArray(aux_shapes[i], ctx);
     if (aux_params.count(aux_names[i]) != 0) {
+      NDArray nd = NDArray(aux_shapes[i], ctx, false, aux_params[aux_names[i]].dtype());
       CopyFromTo(aux_params[aux_names[i]], &nd);
+      aux_arrays.push_back(nd);
+    } else {
+      aux_arrays.push_back(NDArray(aux_shapes[i], ctx));
```
Comment (anirudh2290) [inline src/c_api/c_predict_api.cc]: can you please add a unit test for this in `test_predictor`

**Comment by roywei** [comment]
@kice thanks for your contribution, could you help add a unit test?

**Comment by kice** [comment]
Sorry for the late reply. In fact, I found that after this fix, the GPU memory for INT8 is unusually high (around 25% more compare to FP32). I might consider close this PR until the issue being resolved. I am testing it on Windows 10(17134) with CUDA 9.2 on TTX(Pascal).

**Comment by sandeep-krishnamurthy** [comment]
@kice - Did you get a chance to identify the root cause?

**Comment by sandeep-krishnamurthy** [comment]
@kice - Closing this PR for now due to inactivity. Please reopen it based on your tests and if you think this PR to be reviewed/merged with MXNet.

**Pull request closed without merging**